### DISCUSSION TITLE
Possibility to Access the Activation Level of the Joints

### DISCUSSION THREAD

**Discussion by Nielssss99** [ci-group/revolve2#467] [Ask for help]
Hey

I was wondering whether there is an option to access the outputs of the CPPN over time? If this feature does not exist; where in the code can I find the call to the CPPN during the Mujoco simulation?

Thanks in advance!

Niels

**Comment by oliverweissl** [comment]
If you want to know the control signals, you can looke at `ControlInterfaceImpl`in `mujoco_simulator`. The control interface is then used by the `SimulationHandler`which you can look at in `modular_robot_simulation.ModularRobotSimulationHandler`.

Going through these two classes you will see where the outputs are passed and how they are used.
Additionally you might want to look into `simulators.mujoco_simulator.simulate_scene` to see how they are used in sequence during the simulation.
